src/Fetcher/fetcher_test.go

```diff
@@ -0,0 +1,121 @@
+func TestFetchFilesInvalidArguments(t *testing.T) {
+	// Invalid URL, branch, and specFile arguments.
+	url := "https://invalid-url.com/nonexistent-repo"
+	branch := "branch"
+	specFile := "nonexistent-file.txt"
+
+	err := FetchFiles(url, branch, specFile)
+
+	// We expect an error since the arguments are invalid
+	if err == nil {
+		t.Error("FetchFiles did not return an error when given invalid arguments")
+	} 
+}
+
+func TestFetchFilesValidArguments(t *testing.T) {
+    defer os.Remove("specfile.txt")
+    defer os.RemoveAll("./cache")
+    defer DotGitDelete()
+    
+    //TODO: Maybe this needs to be split into 2 tests
+    
+	// dev repo link and create specfile
+	url := "https://github.com/Agile-Arch-Angels/decentralized-tech-radar_dev.git"
+	//TODO: Change this to main once templates folder is on main
+    branch := "feat_git_fetcher"
+    data := []byte("examples/csv_templates/template.csv")
+    os.WriteFile("./specfile.txt",data,0644)
+    specFile := "specfile.txt"
+
+    err := FetchFiles(url,branch,specFile)
+
+    if err != nil {
+        t.Errorf("FetchFiles returned an err %v", err)
+    }
+
+    _, err = os.Stat("./cache/template.csv")
+   if os.IsNotExist(err) {
+        t.Errorf("File wasn't downloaded or wasn't moved correctly: %v",err)
+    } 
+
+    //TODO: Maybe add a test for contents of CSV file?
+}
+
+
+func TestListingReposForFetch(t *testing.T) {
+    defer os.RemoveAll("./README.md")
+    defer os.RemoveAll("./TestingTextFile.txt")
+    defer os.RemoveAll("./cache")
+    defer DotGitDelete()
+    
+    // Creates a txt file for testing the  
+    textFile, errCreate := os.Create("./TestingTextFile.txt")
+    if errCreate != nil {
+        t.Errorf(errCreate.Error() + " : Couldnt create txt file for testing : TestListingReposForFetch")
+    } 
+    
+    
+    _, errWrite := textFile.WriteString("/README.md")
+    if errWrite != nil {
+        t.Errorf(errWrite.Error() + " : Couldnt write to txt file for testing : TestListingReposForFetch")
+    }
+   
+
+
+    url := "https://github.com/NovoNordisk-OpenSource/decentralized-tech-radar"
+	branch := "main"
+	specFile := "./TestingTextFile.txt"
+
+    url2 := "https://github.com/NovoNordisk-OpenSource/backstage"
+	branch2 := "master"
+	specFile2 := "./TestingTextFile.txt"
+
+    url3 := "https://github.com/NovoNordisk-OpenSource/decentralized-tech-radar"
+	branch3 := "main"
+	specFile3 := "./TestingTextFile.txt"
+
+    var repos []Repo
+    repo := Repo{url, branch, specFile}
+    repo2 := Repo{url2, branch2, specFile2}
+    repo3 := Repo{url3, branch3, specFile3}
+    
+    repos = append(repos, repo, repo2, repo3)
+
+    err := ListingReposForFetch(repos)
+    textFile.Close()
+
+    //check if the error message is as expected.
+    expectedErrorMessage := "failed at fetcher"
+    if  err != nil && strings.Contains(err.Error(), expectedErrorMessage) {
+        os.RemoveAll("./TestingTextFile.txt")
```

Comment:
The reason for the extra steps due to running into ghost .git folders after running the program and running into exceptions. So think it got added as an extra layer of security as this folder is hidden so if a user runs into it then it would be a nightmare to troubleshoot

Reply:
I'll echo what @Slug-Boi says here, it's a securing measure to make sure the .git folders are deleted. However, the code might change in later editions so let's stick with this for now.

Reply:
If there is a reason for it to be there then fine by me. Was more just worried it was pointless code